examples/access_traits/example_cuda_access_traits.cpp

```diff
@@ -75,29 +75,25 @@ int main(int argc, char *argv[])
 
   cudaStream_t stream;
   cudaStreamCreate(&stream);
-  cudaMemcpyAsync(d_a, a.data(), sizeof(a), cudaMemcpyHostToDevice, stream);
+  Kokkos::push_finalize_hook([&stream]() { cudaStreamDestroy(stream); });
```

Comment:
Actually we need to capture it by value

Reply:
```suggestion
  Kokkos::push_finalize_hook([stream]() { cudaStreamDestroy(stream); });
```

Reply:
Note to myself: we need by copy because the stream may already be out of scope by the time this is called.